proposals/dao/fip_999.ts

```diff
@@ -0,0 +1,74 @@
+import { ethers } from 'hardhat';
+import chai, { expect } from 'chai';
+import CBN from 'chai-bn';
+import {
+  DeployUpgradeFunc,
+  NamedContracts,
+  SetupUpgradeFunc,
+  TeardownUpgradeFunc,
+  ValidateUpgradeFunc
+} from '../../types/types';
+import { getImpersonatedSigner } from '@test/helpers';
+
+chai.use(CBN(ethers.BigNumber));
+
+export const deploy: DeployUpgradeFunc = async (deployAddress, addresses, logging = false) => {
+  if (!addresses.core) {
+    console.log(`core: ${addresses.core}`);
+
+    throw new Error('An environment variable contract address is not set');
+  }
+
+  // Create a new Balancer deposit for the BAL/WETH pool
+  const balancerDepositWeightedPoolFactory = await ethers.getContractFactory('BalancerPCVDepositWeightedPool');
+  const balancerDepositBalWeth = await balancerDepositWeightedPoolFactory.deploy(
+    addresses.core,
+    addresses.balancerVault,
+    addresses.balancerRewards,
+    '0x5c6ee304399dbdb9c8ef030ab642b10820db8f56000200000000000000000014', // poolId
+    '300', // max 3% slippage
+    '0xba100000625a3754423978a60c9317c58a424e3D', // BAL token
+    [addresses.balUsdCompositeOracle, addresses.chainlinkEthUsdOracleWrapper]
```

Comment:
have you verified the oracle ordering both in the array and price reporting?

Reply:
yes, will make it more clear with additional tests / invariant checks in validation step